aws-api/src/main/java/com/amplifyframework/api/aws/AppSyncPage.java

```diff
@@ -0,0 +1,33 @@
+package com.amplifyframework.api.aws;
+
+import androidx.annotation.NonNull;
+
+import com.amplifyframework.api.graphql.GraphQLRequest;
+import com.amplifyframework.api.graphql.GraphQLResponse;
+import com.amplifyframework.api.graphql.Page;
+
+final class AppSyncPage<M> extends Page<M> {
+    private final String NEXT_TOKEN_KEY = "nextToken";
+
+    private GraphQLRequest<Page<M>> request;
+    private String nextToken;
+
+    public AppSyncPage(@NonNull GraphQLResponse<Iterable<M>> response,
+                       @NonNull GraphQLRequest<Page<M>> request,
+                       String nextToken) {
+        super(response);
+        this.request = request;
+        this.nextToken = nextToken;
+    }
+
+    @Override
+    public boolean hasNextPage() {
+        return nextToken != null;
+    }
+
+    @Override
+    public GraphQLRequest<Page<M>> getRequestForNextPage() {
+        this.request.addVariable(NEXT_TOKEN_KEY, nextToken);
```

Comment:
I guess this will mutate the existing request. It'd be better to return a new, immutable copy.

To illustate, consider what happens when:
```
GraphQLRequest request = buildOne();
request.getVaraiables() // state A 
new AppSyncPage(response, request, token).getRequestForNextPage();
request.getVariables(); // state B.
```

Reply:
I added a `GraphQLRequest` copy constructor, and am returning a copy.  The value returned is still NOT immutable though, because GraphQLRequest has public methods that support some mutation (e.g. `putVariable`, `addFragment`)

I think that @drochetti might be refactoring `GraphQLRequest` right now anyway as part of making GraphQLRequest builder/helpers, so we should revisit this after that work is done.